### PR TITLE
Trim surrounding whitespace from URL in inlineLink

### PR DESCRIPTION
**Pull request by aidantwoods**
Fixes https://github.com/erusev/parsedown-extra/issues/103

### CODE REVIEW

**Comment by kminchev** [comment]
Why this PR is in this repo?

**Comment by aidantwoods** [comment]
@kminchev 

https://github.com/erusev/parsedown-extra/pull/104#issuecomment-274251222

Also apparently Commonmark reference parser produces the expected result from that issue – so this is really a bug in Parsedown 😉

File: Parsedown.php [hunk 1]
```diff
@@ -1217,7 +1217,7 @@ protected function inlineLink($Excerpt)
             return;
         }
 
-        if (preg_match('/^[(]((?:[^ ()]++|[(][^ )]+[)])++)(?:[ ]+("[^"]*"|\'[^\']*\'))?[)]/', $remainder, $matches))
+        if (preg_match('/^[(]\s*+((?:[^ ()]++|[(][^ )]+[)])++)(?:[ ]+("[^"]*"|\'[^\']*\'))?\s*[)]/', $remainder, $matches))
```
Comment (erusev) [inline Parsedown.php]: what's the role of the `+` in 
image

Comment (aidantwoods) [inline Parsedown.php]: Possessive quantifier – probably not necessary here actually, I tend to use them by habit if I know backtracking won't be needed

**Pull request merged**